src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        public LinterRuleBase(string code, string description, string docUri,
+                          DiagnosticLevel diagnosticLevel = DiagnosticLevel.Warning,
+                          DiagnosticLabel? diagnosticLabel = null)
+        {
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+            this.DiagnosticLevel = diagnosticLevel;
+            this.DiagnosticLabel = diagnosticLabel;
+        }
+
+        internal const string FailedRuleCode = "Linter Rule Error";
+        private IConfigurationRoot? Config;
+        public string AnalyzerName { get; }
+
+        public string Code { get; }
+        public readonly string RuleConfigSection = $"{LinterAnalyzer.SettingsRoot}:{LinterAnalyzer.AnalyzerName}:rules";
+        public bool Enabled => this.DiagnosticLevel != DiagnosticLevel.Off;
+        public Diagnostics.DiagnosticLevel DiagnosticLevel { get; private set; }
+        public string Description { get; }
+        public string DocumentationUri { get; }
+        public Diagnostics.DiagnosticLabel? DiagnosticLabel { get; }
+
+        /// <summary>
+        /// Override to implement detailed message for rule
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        protected virtual string FormatMessage(params object[] values) => this.Description;
+
+        public virtual void Configure(IConfigurationRoot config)
+        {
+            this.Config = config;
+            var configDiagLevel = GetConfiguration(nameof(this.DiagnosticLevel).ToLower(), this.DiagnosticLevel.ToString());
+            if (DiagnosticLevel.TryParse<DiagnosticLevel>(configDiagLevel, true, out var lvl))
+            {
+                this.DiagnosticLevel = lvl;
+            }
+        }
+
+        /// <summary>
+        /// Gets a message using the supplied parameter values (if any).
+        /// Otherwise returns the rule description
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        internal string GetMessage(params object[] values)
+        {
+            return (values.Any() ? FormatMessage(values) : this.Description)
+                + Environment.NewLine
+                + string.Format(CoreResources.SeeDocLinkFormat, this.DocumentationUri);
+        }
+
+        public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            try
+            {
+                // Expand the iteration immediately or the try/catch won't catch exceptions occuring during the rule analysis
+                // TODO: We need exception handling further up the tree in order to handle external linters.
+                return AnalyzeInternal(model).ToArray();
+            }
+            catch (Exception ex)
+            {
+                return new[]{ new AnalyzerDiagnostic(this.AnalyzerName,
```

Comment:
>AnalyzerDiagnostic [](http://example.com/codeflow?start=34&length=18)

The analyzer looks for classes deriving from the interface instead of this base class. So if a rule doesn't derive from the base class, this hardening will basically be skipped. 

Shouldn't we centralize this sort of logic in the analyzer so it's indepedent of the rule implementation?

Reply:
Yes, thus this comment:

// TODO: We need exception handling further up the tree in order to handle external linters.

Still nice to have it here for core rules to have better granularity on errors.

Reply:
Didn't see the TODO. Makes sense.

Reply:
That's on the list for when 3rd-party linters are designed.